src/main/java/org/reaktivity/nukleus/http2/internal/routable/stream/SourceInputStreamFactory.java

```diff
@@ -152,7 +152,7 @@ public MessageHandler newStream()
         private final Int2ObjectHashMap<Http2Stream> http2Streams;
         private int lastPromisedStreamId = 0;
 
-        private final AtomicBuffer slab = new UnsafeBuffer(new byte[4096]);
+        private AtomicBuffer slab;
```

Comment:
To avoid unnecessary overhead per stream, the slab should be shared by all streams created by the same stream factory, and acquired for a specific stream only when needed - see `nukleus-http.java` for a working example.